pynuodb/encodedsession.py

```diff
@@ -188,6 +188,13 @@ def check_auth(self):
         except SessionException as e:
             raise ProgrammingError('Failed to authenticate: ' + str(e))
 
+    def get_auth_types(self):
```

Comment:
Because this is the first message that is send to the TE. You do not know the protocol version the server is running. Correct me if I am mistaken please and the broker gives you the protocol version along with ip/port.